### PR TITLE
ENH: support anaconda labels

### PR DESCRIPTION
**Pull request by ebolyen**

### CODE REVIEW

File: assets/common.py [hunk 1]
```diff
@@ -146,6 +160,7 @@ def ftp_lock(ftp):
     locked = False
     while not locked:
         try:
+
```
Comment (thermokarst) [inline assets/common.py]: ?

File: assets/common.py [hunk 2]
```diff
@@ -130,12 +135,21 @@ def download(self, path, filehandle):
...
-        subprocess.run('anaconda upload --force -u %r %s'
-                       % (self._channel, ' '.join(files_to_upload)),
-                       shell=True, check=True, stdout=subprocess.DEVNULL,
-                       stderr=subprocess.DEVNULL)
+        files = ' '.join(repr(os.path.join(data.root, relpath))
+                         for relpath in relpaths)
+        with tempfile.TemporaryFile('w+') as stdout, \
+                tempfile.TemporaryFile('w+') as stderr:
+            try:
+                subprocess.run('anaconda upload -u %r -l %r %s'
```
Comment (thermokarst) [inline assets/common.py:143]: 👍

File: assets/out [hunk 3]
```diff
@@ -21,12 +21,16 @@ def main(source, params, inpath):
...
-        relpaths = conn.upload_local_data(data, pkg_name, pkg_version)
+        remote_entries = common.ChannelData(conn=conn).iter_entries(
+            pkg_name, pkg_version)
+        if not list(remote_entries):
+            relpaths = conn.upload_local_data(data, pkg_name, pkg_version)
 
     return {'version': common.to_version(pkg_version),
-            'metadata': [{'name': 'file_%d' % idx, 'value': relpath}
-                         for idx, relpath in enumerate(relpaths)]}
+            'metadata': [{'name': 'file', 'value': relpath}
```
Comment (thermokarst) [inline assets/out:32]: 🎉

**Review by thermokarst — APPROVED** [review]
LGTM!

File: assets/common.py (hunk 1 above)
Comment (ebolyen) [reply assets/common.py]: oh, that's strange

**Pull request merged**